### ISSUE TITLE
Fix uninstaller

### ISSUE THREAD

**Issue by wshawn** [Ibochkarev/miniShop2#49]
Currently, the removal of your Package leaves all of the tables in the database and the elements in the Resource Tree. This is not the expected functionality of remove.

All of these should be removed when your package is removed.

**Comment by nnmer** [comment]
you mean uninstall or remove?
also, what if i usually remove old version of components....if it will remove tables - i don't think people will be happy to use this extra.

we can compare with OS.
Applications in Windows and Linux both leaves settings and it's DB if any. and if you again install the application - then you are able to use  your data.

**Comment by bezumkin** [comment]
Show me the package, that deletes all user-generated content including tables - i want to see it.

Articles, Gallery, maybe Quip do that? I do not think so.

**Comment by wshawn** [comment]
If you were using a prefix it wouldn't be so messy.  You are also assuming there is information in the tables.

Pay close attention to the difference of uninstall (which leaves the database alone) and remove....

**Comment by wshawn** [comment]
Also your package is filling the error log with :
[2013-07-06 19:33:35](ERROR @ /connectors/resource/index.php) Could not load class: msCategory from mysql.mscategory.

After the Package is Removed, the tables manually deleted, sessions flushed, and the core/cache folder manually deleted those errors are coming up.

**Comment by wshawn** [comment]
Errors from filesystem:
[11-Jun-2013 15:51:03 America/Chicago] PHP Fatal error:  Call to a member function get() on a non-object in /home/zcellula/public_html/manager/controllers/default/element/chunk/update.class.php on line 132
[11-Jun-2013 15:51:52 America/Chicago] PHP Fatal error:  Call to a member function get() on a non-object in /home/zcellula/public_html/manager/controllers/default/element/chunk/update.class.php on line 132

**Comment by wshawn** [comment]
Traces left in the database after Removal:
2 matches inside table modx_lexicon_entries
2 matches inside table modx_site_content  <=---  this is probably the issue,  I will switch the class key back to modDocument
1 match inside table modx_user_settings <=- actually there was also a ms_product value there also

I removed the offending entries and the errors stopped.

I took the time to document it here so the next person having these same issues will know how to clean miniShop2 completely from their site if they so desire.

Have a nice day.
